yext/answers-hitchhiker-theme
# Add a 'development' mode toggle to the Webpack configuration.

This PR allows our Webpack toolchain to be easily toggled between two modes:
'production' and 'development'. These are built-in Webpack modes. In
'development', minimization of JS assets process by Webpack does not occur. We
add the additional behavior that bundle.js and bundle.css are not inlined
in 'development'. This inlining behavior is not too important during local
development. Mainly, it offers a performance boost for published pages. But, it
considerably slows down the Webpack compilation.

By running Webpack in 'development' and getting rid of inlining, preview
generation should happen much faster and use considerably less memory. I tried
the new 'development' mode for YAnswers. I found that memory usage reduced by
60% and the compilation time reduced by about 70%.

Eventually, we will get an environment variable passed to our toolchain that
will toggle this behavior. In SGS, users will be able to pick between the two
modes. Publishes will use only 'production' mode.

TEST=manual

This PR was tested using the YAnswers site. I ran the following scenarios:
- Set isProduction to false. Verified that Webpack built the assets in the
  'development' mode and no inlining occurred. Spot checked pages on the site
  to ensure assets were generated without issue.
- Set isProduction to true. Verified that Webpack built the assets in the
  'production' mode. Inlining of bundle.js and bundle.css occurred as exepcted.
  Spot checked some pages to ensure things all looked/worked correctly.
